pkg/controller/core/workload_controller.go

```diff
@@ -571,7 +595,7 @@ func (w *workloadCqHandler) Update(ctx context.Context, ev event.UpdateEvent, wq
 	log.V(5).Info("Workload cluster queue update event")
 	oldCq, oldIsQueue := ev.ObjectOld.(*kueue.ClusterQueue)
 	newCq, newIsQueue := ev.ObjectNew.(*kueue.ClusterQueue)
-	if oldIsQueue && newIsQueue && !slices.CmpNoOrder(oldCq.Spec.AdmissionChecks, newCq.Spec.AdmissionChecks) {
+	if oldIsQueue && newIsQueue && (!slices.CmpNoOrder(oldCq.Spec.AdmissionChecks, newCq.Spec.AdmissionChecks) || !newCq.DeletionTimestamp.IsZero()) {
```

Comment:
for readability, it's useful to have something like: `if !oldIsQueue || !newIsQueue { return }`.

It makes it clear that we are exiting because basic checks are not satisified.

But since it's just one line after the check, this is acceptable.

Reply:
done